assignments/hof-implementation/code.js

```diff
@@ -1,12 +1,30 @@
 // N.B. Do not use the array methods of the same name to implement these functions!
+// i dont know what you mean by this
 
 const filter = (predicate, array) => {
+  let product = []
+  for (let i = 0; i < array.length; i++) {
+    if (predicate(array[i]) === true) {
```

Comment:
You don't need to compare a boolean value (such as that returned by `predicate`) to a literal as it's already a boolean. So you can just write:

```suggestion
    if (predicate(array[i])) {
```